### PR TITLE
TCP tests with utils

### PR DESCRIPTION
**Pull request by katrybacka**

### CODE REVIEW

**Comment by katrybacka** [comment]
recheck no bug

**Comment by katrybacka** [comment]
recheck no bug

File: CIScripts/DockerFiles/iis-tcptest/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM microsoft/iis
+
+RUN mkdir C:\TCPTest
+RUN echo "<h1>Hello, Contrail!</h1>" > C:\TCPTest\index.html
```
Comment (jacek-izykowski) [inline CIScripts/DockerFiles/iis-tcptest/Dockerfile]: This is not correct HTML (missing DOCTYPE, etc.)...

File: CIScripts/Test/Utils/IisTcpTestDockerImageBuild.ps1 [hunk 2]
```diff
@@ -0,0 +1,19 @@
+. $PSScriptRoot\..\..\Common\Aliases.ps1
+$DockerfilePath = "$PSScriptRoot\..\..\DockerFiles\iis-tcptest\Dockerfile"
+
+function Initialize-IisTcpTestDockerImage  {
```
Comment (PawelMarchewkaCodiLime) [inline CIScripts/Test/Utils/IisTcpTestDockerImageBuild.ps1]: [Refinement] We think that this function could be more generic and used for any docker image. Image name could be passed as parameter. In this case it would be 'iis-tcptest'. This function can assume that all docker files are present in $PSScriptRoot\..\..\DockerFiles\.

File: CIScripts/Test/Utils/IisTcpTestDockerImageBuild.ps1 [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session)
+
+    $DockerFilesDir = "C:\DockerFiles"
+    Invoke-Command -Session $Session -ScriptBlock {
+        New-Item -ItemType Directory -Force $Using:DockerImagesDir | Out-Null
+    }
+
+    Write-Host "Copying iis-tcp-test image Dockerfile"
+    Copy-Item -ToSession $Session -Path $DockerfilePath -Destination $DockerFilesDir
```
Comment (PawelMarchewkaCodiLime) [inline CIScripts/Test/Utils/IisTcpTestDockerImageBuild.ps1]: We think that this should copy not only dockerfile but whole directory 'iis-tcptest'

File: CIScripts/Test/Utils/IisTcpTestDockerImageBuild.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    [Parameter(Mandatory=$true)] [string] $TestenvConfFile
+)
+
+. $PSScriptRoot\IisTcpTestDockerImageBuild.ps1
+
+$Sessions = New-RemoteSessions -VMs (Read-TestbedsConfig -Path $TestenvConfFile)
+$Session = $Sessions[0]
+
+Describe "Initialize-IisTcpTestDockerImage" {
+    It "Builds iis-tcptest image" {
```
Comment (PawelMarchewkaCodiLime) [inline CIScripts/Test/Utils/IisTcpTestDockerImageBuild.Tests.ps1]: If you decide to make Initialize-IisTcpTestDockerImage more generic, please also test negative case with invalid docker image name.

File: CIScripts/Test/Tests/Protocol/SingleHostWithUtils.Tests.ps1 [hunk 5]
```diff
@@ -129,6 +148,8 @@ Describe "Single compute node protocol tests with utils" {
 
 
     BeforeAll {
+        Initialize-IisTcpTestDockerImage -Session $Session
```
Comment (PawelMarchewkaCodiLime) [inline CIScripts/Test/Tests/Protocol/SingleHostWithUtils.Tests.ps1]: [Refinement] Maybe we should remove this docker image in AfterAll

File: CIScripts/Test/Tests/Protocol/SingleHostWithUtils.Tests.ps1 [hunk 6]
```diff
@@ -118,9 +101,45 @@ Describe "Single compute node protocol tests with utils" {
...
+        $Container1NetInfo = Get-RemoteContainerNetAdapterInformation `
+            -Session $Session -ContainerID $Container1ID
+        $Container2NetInfo = Get-RemoteContainerNetAdapterInformation `
+            -Session $Session -ContainerID $Container2ID
+
+        Initialize-ContainersConnection -VMNetInfo $VMNetInfo -VHostInfo $VHostInfo `
+            -Container1NetInfo $Container1NetInfo -Container2NetInfo $Container2NetInfo `
+            -Session $Session
+
+        Write-Host "Executing netsh"
```
Comment (PawelMarchewkaCodiLime) [inline CIScripts/Test/Tests/Protocol/SingleHostWithUtils.Tests.ps1]: This code doesn't seem to have an equivalent in removed/changed code. Is there some piece of dead code somewhere that was responsible for setting up arp table?

File: CIScripts/Test/Tests/Protocol/SingleHostWithUtils.Tests.ps1 [hunk 7]
```diff
@@ -118,9 +101,45 @@ Describe "Single compute node protocol tests with utils" {
             -TenantName $ControllerConfig.DefaultProject `
             -Name $NetworkName `
             -Subnet "$( $Subnet.IpPrefix )/$( $Subnet.IpPrefixLen )"
+
+        Write-Host "Creating containers"
+        $Container1ID, $Container2ID = Invoke-Command -Session $Session -ScriptBlock {
+            docker run --network $Using:NetworkName -d iis-tcptest ping -t localhost
```
Comment (PawelMarchewkaCodiLime) [inline CIScripts/Test/Tests/Protocol/SingleHostWithUtils.Tests.ps1]: [Change] In case of 'iis-tcptest' I believe 'ping -t localhost' is not required.

**Comment by katrybacka** [comment]
recheck no bug

File: CIScripts/Test/Tests/Protocol/SingleHostWithUtils.Tests.ps1 (hunk 5 above)
Comment (krdln) [reply CIScripts/Test/Tests/Protocol/SingleHostWithUtils.Tests.ps1]: cc @jacek-izykowski 

I strongy disagree with removing the image in `AfterAll` as it prevents docker from caching the image. This slows down both manual testing and testing in CI (when there'll be multiple tests using this image).

Without the removal, the tests are still independent, the only side effect is that `Initialize-DockerImage` will sometimes be faster.
